pkg/apis/query/v0alpha1/query.go

```diff
@@ -75,6 +80,21 @@ type GenericDataQuery struct {
 	props map[string]any `json:"-"`
 }
 
+type ResultDataContract struct {
+	// Type asserts that the frame matches a known type structure.
+	Type data.FrameType `json:"type,omitempty"`
+
+	// TypeVersion is the version of the Type property. Versions greater than 0.0 correspond to the dataplane
+	// contract documentation https://grafana.github.io/dataplane/contract/.
+	TypeVersion data.FrameTypeVersion `json:"typeVersion"`
+
+	// Maximum bytes that can be read -- if the query planning expects more then this, the query may fail fast
+	MaxBytes int64 `json:"maxBytes,omitempty"`
+
+	// Maximum labels to expand -- if the query planning expects more then this, the query may fail fast
```

Comment:
What does expand mean? The number of items (e.g. labeled numbered, labeled time series?)

Reply:
good question -- need to sort out the language, added it here as a placeholder/stub.

The main concern (IIUC) is from alerting where you write a query that initially resolves to some small set of labels and later becomes a huge set of labels that then trigger behavior.

This may be better represented as maxFrames?  since loki/prometheus will create a new frame for each label set